pex/bin/pex.py

```diff
@@ -503,11 +503,18 @@ def make_relative_to_root(path):
   return os.path.normpath(path.format(pex_root=ENV.PEX_ROOT))
 
 
-def main():
+def main(args=None, log_callback=None):
+  if log_callback is not None:
```

Comment:
I'm not sure I understand your statements or reasoning here.

"I'd rather not put mutable objects in as the default options, which is why I went with None." -> what default option here is mutable? `log` is a function defined above. `sys.exit` is also a function. if anything, the whole 'log can be either be a global or local depending on the value of log_callback' strikes me as most peculiar.

"As far as the sys.argv[1:] being placed in main I'd rather not put that logic somewhere it can't be easily tested." -> somehow I feel like the slicing of `sys.argv` isn't a super critical thing to test? also, by precedent injection of args is actually a super common pattern in `twitter.common.app`. what's the theoretical harm here in your mind? or why not alternatively `main(sys.argv)`?

I'm a firm -1 on avoiding needless clutter in `main()` for tests and particularly not a fan of this approach as-is - would prefer that you either clean this up in-line with the suggestions or find an alternate route.